### ISSUE TITLE
Incorrect handling of '\n'

### ISSUE THREAD

**Issue by ioquatix** [whitequark/parser#547]
```ruby
require 'parser/current'

module Rewriter
  def self.rewrite(path, code = nil)
    code ||= File.read(path)

    ast = Parser::CurrentRuby.parse(code)

    buffer = Parser::Source::Buffer.new(path)
    buffer.source = code

    rewriter = Rewriter.new

    # Rewrite the AST, returns a String with the new form.
    rewriter.rewrite(buffer, ast)
  end

  class Rewriter < ::Parser::TreeRewriter
  end
end

code = Rewriter.rewrite("<test>", <<~EOF)
def lines(things)
  things.join('\n')
end
EOF

puts code
```

The `'\n'` string is incorrectly rewritten.

Expected output, same as input. Actual output:

```
def lines(things)
  things.join('
')
end
```

**Comment by iliabylich** [comment]
It doesn't look like a bug, that's how heredocs parse `\n`:
``` ruby
$ /bin/cat test.rb
puts <<~EOF
def lines(things)
  things.join('\n')
end
EOF

$ ruby -v test.rb
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin18]
def lines(things)
  things.join('
')
end
```

**Comment by ioquatix** [comment]
Hmm, I had this bug in a different situation but tried to make a small repro - let me try again.

**Comment by ioquatix** [comment]
Okay, my bad. It turns out minitest was trying to reformat the output and it looked wrong but it was actually okay.

```
     measure{respond_with @clients.each(&formatter).join('\
-n'), content_type: 'application/json; boundary=NL'}
+'), content_type: 'application/json; boundary=NL'}
```

Not really sure why it inserts a newline there.
